lib/bson/dbref.rb

```diff
@@ -84,7 +84,7 @@ def initialize(hash)
         end
       end
 
-      super(hash)
+      super(reorder_fields(hash))
```

Comment:
This needs to be done before the validation so that symbol keys are accepted (e.g. `:$ref`)

Reply:
done